### ISSUE TITLE
Wrong email or password

### ISSUE THREAD

**Issue by karepiu** [edc1591/python-winix#2]
Hi, 

I have Winix Smart app (android) and Winix C545 air purifier. When trying to use your code I get error. From debugging it seems it is simply wrong password or email. Did encryption key changed recently or I need some other account/app ?

Thanks

**Comment by steveb85** [comment]
I think the address of the server changed, I’m currently working on a build for the am90, but not sure about the c545

**Comment by karepiu** [comment]
Looks like it is same app - Winix Smart App. So it should work. Do you have new address ?

**Comment by steveb85** [comment]
Yes, but the structure of the commands is very different, I’m currently working through that, it’s my first attempt, very happy to share if you have skills, and see what we can get to? I can currently control and get a status update via CLI, but now trying to sort out the logic and get it into a hb plugin, based, largely off what has been done on this plugin.

**Comment by karepiu** [comment]
Yes please share. I want to integrate it with Home Assistant. If you have it in Python great, if not I will rewrite from any language you had it in. Thanks

**Comment by steveb85** [comment]
https://github.com/evandcoleman/python-winix here is the library i found in python, it has most of the bits I'm sure you'll need, i can share the address, and the auth is just what you use in the app, and i can share how the status works / the control codes. Sounds like you've got a lot more experience than i do!

**Comment by karepiu** [comment]
This link points to this repo (this discussion is in issue associated with very same repo you linked). Can you post you findings here or if you have working code can just commit to your repo and provide a link so I could take a look.

**Comment by steveb85** [comment]
Sorry my bad, copied the wrong one. https://github.com/banzalik/homebridge-winix-c545 Is the one i meant to link to. I’m about to do a bit of trial and error and will copy a link to what i update tonight.

**Comment by steveb85** [comment]
hey @karepiu here is my super rough workings. please excuse my terrible work, i tried to add the explanations in as much as possible. https://github.com/steveb85/axiostest I've got it to read the server, return the status, and i'm figured out what the each part of the status is (apart from the air sensor, which i figured i'd attack when i could at least get the next part done) 

hopefully there is some stuff to help you in there. happy to move this over there if it's helpful at all.

**Comment by karepiu** [comment]
Did you commit everything. I just see single login request in your code. 
Looks like in your repo they use same approach as here - [https://github.com/hfern/winix]

If hfern code works I should have everything I need.

**Comment by steveb85** [comment]
You’re right I committed but didn’t push the updates, it should be done now. I’ll check out what you sent, sorry about that.

**Comment by steveb85** [comment]
@karepiu how’d you go? Sorted?

**Comment by karepiu** [comment]
@steveb85 Yep. Quite simple code actually. No issues at all. Already have full device control and status code in python, working on device discovery now and then will link it to home assistant ( could link it to home assistant without discovery but want to make full solution ). Should have full working purifier in home assistant by end of week. Once I finish the python part if times allows I will rewrite it to node.js module or lib ( javascript ).

If you have any questions happy to answer them. 

TIP: If you are trying to decode the status values - forget the javascript and axcios. Use Postman/Fiddler or in this case as status/control is GET based just use your browser.

**Comment by steveb85** [comment]
@karepiu cool, glad you sorted it. I may have reached my end on this. I’ll see if you get it into node otherwise next week i may see if i can try out home assistant on the same pi to get it going, not sure if it will work with homebridge as well. I’ve learned a lot, but am hitting a wall, and i think you’re right, axios isn’t the best method for all this.

**Comment by karepiu** [comment]
Can you just open this in your browser ( substitute device_id for your own ) and put the result here ( please remove device from response ). Does this link work for you at all ?

https://us.api.winix-iot.com/common/event/sttus/devices/[YOUR DEVICE ID]

**Comment by steveb85** [comment]
Thanks @karepiu 

Result is :

{"statusCode":200,"headers":{"resultCode":"F300","resultMessage":"parameter(s) not valid : device id"},"body":{}} 

When i used mitmproxy that’s how i got the info i got, and found what i needed to control it. 

I can control my winix via cli with axios, and i even decoded the status Json reply, but that’s where i get stumped.

**Comment by karepiu** [comment]
So looks like you device uses different service. 
If you decoded the status and have ability to control it means you have all. Missing part ( based on your comment in the other issue ) is connecting it to homebridge, but you also have example how to do it in C545 homebridge code that you linked. 

Basically the simplest way is to download and modify the C545 homebridge repo. Replace urls with correct for your device. Modify the codes if different than for C545 and that is it. Start with functionalities that are mosts likely shared - on/off, speed. Ignore the rest. Add it to your homebridge and check if it works. 

Not sure where is the problem - wanna elaborate more ?

**Comment by steveb85** [comment]
The big difference is that the site for the am90 needs authentication and it needs to be a “post” with a body message containing certain information. That i can do, but then the status is returned as one section in the body return. I can do all this through CLI with axios, and am getting closer, but I’m getting lost in the part where i have to take the information from the body of the reply and have the application understand what it means, even though i now can. It’s all very new to me, and surely with enough time I’ll figure something out, especially with all the information on the internet.
